src/install/deploySteps.js

```diff
@@ -0,0 +1,221 @@
+
+  const startApps = () => {
+    info('Starting all apps…', apps.APPS);
+    return apps.start()
+      .then(() => info('All apps started.'));
+  };
+
+  const loadStagedDdocs = () => {
+    return utils.getStagedDdocs(true, true)
+      .then(stagedDdocs => {
+        const stagedMainDdocId = utils.getStagedDdocId(utils.mainDdocId(deployDoc));
+
+        return stagedDdocs.reduce((acc, ddoc) => {
+          if (ddoc._id === stagedMainDdocId) {
+            acc.primaryDdoc = ddoc;
+          } else {
+            acc.secondaryDdocs.push(ddoc);
+          }
+
+          return acc;
+        }, {secondaryDdocs: []});
+      });
+  };
+
+  const deploySecondaryDdocs = secondaryDdocs => {
+    debug(`Secondary ddocs: ${secondaryDdocs.map(d => d._id)}`);
+    debug('Getting currently deployed secondary ddocs');
+
+    const deployIds = secondaryDdocs
+      .map(ddoc => ddoc._id)
+      .map(utils.getDeployedDdocId);
+
+    return DB.app.allDocs({keys: deployIds})
+      .then(({rows: deployedStubs}) => {
+        debug(`Found ${deployedStubs.length}`);
+
+        secondaryDdocs.forEach(ddoc => {
+          ddoc._id = utils.getDeployedDdocId(ddoc._id);
+
+          const currentlyDeployed = deployedStubs.find(d => d.id === ddoc._id);
+          if (currentlyDeployed) {
+            debug(`${ddoc._id} already exists, overwriting`);
+            ddoc._rev = currentlyDeployed.value.rev;
+          } else {
+            debug(`${ddoc._id} is new, adding`);
+            delete ddoc._rev;
+          }
+        });
+
+        debug('Writing secondary ddocs');
+        return DB.app.bulkDocs(secondaryDdocs)
+          .then(results => debug(`Secondary ddocs written: ${JSON.stringify(results)}`));
+      });
+  };
+
+  const deployPrimaryDdoc = primaryDdoc => {
+    debug(`Primary ddoc: ${primaryDdoc._id}`);
+    debug('Checking to see if primary exists already');
+
+    primaryDdoc._id = utils.getDeployedDdocId(primaryDdoc._id);
+
+    return DB.app.get(primaryDdoc._id)
+      .catch(err => {
+        if (err.status !== 404) {
+          throw err;
+        }
+      })
+      .then(deployedDdoc => {
+        if (deployedDdoc) {
+          debug('It does, copying config to staged ddoc before writing');
+          primaryDdoc.app_settings = deployedDdoc.app_settings;
+          primaryDdoc._rev = deployedDdoc._rev;
+        } else {
+          debug('It does not');
+          delete primaryDdoc._rev;
+        }
+
+        debug('Writing primary ddoc');
+        return DB.app.put(primaryDdoc)
+          .then(() => debug('Primary ddoc written'));
+      });
+  };
+
+  const deployStagedDdocs = () => {
+    info(`Deploying staged ddocs`);
+
+    return moduleWithContext._loadStagedDdocs()
+      .then(({primaryDdoc, secondaryDdocs}) => {
+        return moduleWithContext._deploySecondaryDdocs(secondaryDdocs)
+          .then(() => moduleWithContext._deployPrimaryDdoc(primaryDdoc));
+      });
+  };
+
+const updateSymlinkAndRemoveOldVersion = changedApps =>
```

Comment:
I really don't like using the no brace style for > 1 line functions. It's really unclear (to me) that anything is being returned. Do you feel strongly about this or can we use braces for functions longer than 1 line?

Reply:
I think it looks worse, but I don't _strongly_ think it looks worse. Especially stuff like:

I'll try to break my programming and add braces

Reply:
I think it looks worse, but I don't _strongly_ think it looks worse. Especially stuff like:

```js
const foo = () => {
  return doStuff
    .then(doMoreStuff)
    .then(evenMoooooore)
    .then(results => {
    }
};
```

Compared to:

```js
const foo = () =>

I'll try to break my programming and add braces